### PR TITLE
fix: extend openapi meta collected from domain models

### LINKED ISSUE #3232: Bug: return DTO causes field descriptions in Pydantic model to disappear
### Description

When using a Pydantic model without a return DTO, the field descriptions show up correctly in /schema documentation. With a return_dto, field descriptions disappear.

## No DTO - Field Descriptions Present
_See bottom portion of image below data types for `id` and `user_name` ._
No_DTO_Field_descriptions_present

## With return DTO - Field Descriptions now missing
Return_DTO_Field_descriptions_missing

### URL to code causing the issue

_No response_

### MCVE

```python
from typing import Annotated

from litestar import Litestar, get
from litestar.contrib.pydantic import PydanticDTO
from litestar.dto import DTOConfig
from pydantic import BaseModel, Field


class User(BaseModel):
    id: Annotated[
        int,
        Field(description="This is a test (id description)."),
    ]

    user_name: str = Field(description="Not annotated test (name description)")


class DataCollectionDTO(PydanticDTO[User]):
    config = DTOConfig(rename_strategy="camel")


USER_DB = {1: {"id": 1, "user_name": "Stella"}}


# #1 - NO DTO: field descriptions will be present in /schema
@get("/user/{user_id:int}", sync_to_thread=False)
# #2 - With return_DTO: field descriptions are missing from schema
# @get("/user/{user_id:int}", return_dto=DataCollectionDTO, sync_to_thread=False)
async def get_user(user_id: int) -> User:
    return USER_DB[user_id]


app = Litestar(route_handlers=[get_user])
```

### Steps to reproduce

```bash
1. Build and run a litestar application using the code above for app.py.
2. View the /schema for the API (likely at http://127.0.0.1:8000/schema).
3. See the field descriptions for id and user_name as shown in the top image above. Field descriptions should be present
4. Edit app.py by commenting out the first decorator for get_user() and removing the comment from the second decorator line.
5. Again run the litestar application and view the /schema for the API.
6. Now the field descriptions are missing as in the second image above.
```

### Screenshots

_No response_

### Logs

_No response_

### Litestar Version

Litestar 2.4.2  (running on Ubuntu in WSL2)

### Platform

- [X] Linux
- [ ] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3232">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3232/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by peterschutt**
## Description

`FieldDefinition`s pack any OpenAPI metadata onto a `KwargDefinition` instance when types are parsed from domain models.

When we produce a DTO type, we transfer this meta from the `KwargDefinition` to a `msgspec.Meta` instance, however so far this has only included constraints, not attributes such as descriptions, examples and title.

This PR ensures that we transfer the openapi meta for the complete intersection of fields that exist on both `KwargDefinition` and `Meta`.

## Closes

Closes #3232

### CODE REVIEW

File: litestar/dto/_backend.py [hunk 1]
```diff
@@ -750,14 +750,17 @@ def _create_struct_field_meta_for_field_definition(field_definition: TransferDTO
         return None
 
     return msgspec.Meta(
-        gt=kwarg_definition.gt,
+        description=kwarg_definition.description,
+        examples=[e.value for e in kwarg_definition.examples or []],
```
Comment (peterschutt) [inline litestar/dto/_backend.py:754]: we unpack the examples here for the same reason as #3224

File: litestar/dto/_backend.py [hunk 2]
```diff
@@ -750,14 +750,17 @@ def _create_struct_field_meta_for_field_definition(field_definition: TransferDTO
...
         ge=kwarg_definition.ge,
-        lt=kwarg_definition.lt,
+        gt=kwarg_definition.gt,
         le=kwarg_definition.le,
-        multiple_of=kwarg_definition.multiple_of,
-        min_length=kwarg_definition.min_length if not field_definition.is_partial else None,
+        lt=kwarg_definition.lt,
         max_length=kwarg_definition.max_length if not field_definition.is_partial else None,
+        min_length=kwarg_definition.min_length if not field_definition.is_partial else None,
```
Comment (guacs) [inline litestar/dto/_backend.py:760]: Why is there a need for `is_partial` check and why is it only needed for `min_length` and `max_length`? I didn't quite understand that.

Comment (peterschutt) [reply litestar/dto/_backend.py:760]: This isn't new to this PR - I just sorted them alphabetically so it's showing in the diff.

As to why - I'm not sure either. Maybe something to do with how partial makes the type a union with msgspec's `UnsetType`, but I'm only guessing.

Comment (guacs) [reply litestar/dto/_backend.py:760]: Aah got it. I'll try to look into that separately from this PR.

**Review by guacs — APPROVED** [review]

**Pull request merged**